src/Controller/PageAdminController.php

```diff
@@ -253,16 +253,19 @@ public function composeContainerShowAction(Request $request): Response
         // filter service using the template configuration
         if (null !== $page) {
             $template = $this->container->get('sonata.page.template_manager')->get($page->getTemplateCode());
+            $code = $block->getSetting('code');
 
-            $container = $template->getContainer($block->getSetting('code'));
+            if (null !== $code) {
+                $container = $template->getContainer($code);
 
-            if (isset($container['blocks']) && \count($container['blocks']) > 0) {
-                foreach ($blockServices as $code => $service) {
-                    if (\in_array($code, $container['blocks'], true)) {
-                        continue;
-                    }
+                if (null !== $container) {
```

Comment:
better to chekc with null, $container is an array (with 'blocks' set) or null

Reply:
We could have more test for the compose page (but panther test would do it better since we could test js behavior)